src/scripts/main.js

```diff
@@ -2,4 +2,33 @@
 
 document.addEventListener('click', (e) => {
   // write code here
+
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+  const wallRectangle = wall.getBoundingClientRect();
+
+  let x = e.clientX - wallRectangle.left - spider.width + spider.width / 2;
+
+  let y = e.clientY - wallRectangle.top - spider.height + spider.height / 2;
+
+  if (e.clientX <= wallRectangle.left + spider.width / 2) {
+    x = 0;
+  } else if (
+    e.clientX >=
+    wallRectangle.left + wallRectangle.width - spider.width
+  ) {
+    x = wallRectangle.width - spider.width - spider.width / 2;
```

Comment:
The calculation for `x` when the spider reaches the right boundary should be `wallRectangle.width - spider.width` to prevent it from exceeding the right edge.